Record with static fields throws sad sad panda error

**Prettier-Java 1.1.0**

```sh
# Options (if any):
tabWidth: 4
printWidth: 100
trailingComma: "none"
```

**Input:**

```java
// code snippet
    record MyRecord(int i) {
        static final String STATIC_FINAL = "";
        static String staticNonfinal = "";
    }
```

**Output:**

```java
// something prettier, doesn't really matter
```

**Expected behavior:**
To not throw an exception. [Records are allowed to have static fields](https://docs.oracle.com/en/java/javase/16/language/records.html), but prettier-java throws this error:
(Note that if I add a `public` modifier, prettier-java then starts to complain about that first. Also interesting: both `public` and `static` are in the list of allowed tokens.)

```
[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.11.1:apply (default-cli) on project equalsverifier: Execution default-cli of goal com.diffplug.spotless:spotless-maven-plugin:2.11.1:apply failed: Unexpected response status code at /prettier/format [HTTP 501] -- (Error while formatting: Error: Sad sad panda, parsing errors detected in line: 148, column: 9!
[ERROR] Expecting: one of these possible Token sequences:
[ERROR]   1. ['@']
[ERROR]   2. ['final']
[ERROR]   3. ['byte']
[ERROR]   4. ['short']
[ERROR]   5. ['int']
[ERROR]   6. ['long']
[ERROR]   7. ['char']
[ERROR]   8. ['float']
[ERROR]   9. ['double']
[ERROR]   10. ['boolean']
[ERROR]   11. [Identifier]
[ERROR]   12. ['@']
[ERROR]   13. ['public']
[ERROR]   14. ['protected']
[ERROR]   15. ['private']
[ERROR]   16. ['abstract']
[ERROR]   17. ['static']
[ERROR]   18. ['final']
[ERROR]   19. ['strictfp']
[ERROR]   20. ['class']
[ERROR]   21. ['enum']
[ERROR]   22. ['record']
[ERROR]   23. ['{']
[ERROR]   24. ['yield']
[ERROR]   25. [';']
[ERROR]   26. ['(']
[ERROR]   27. [Identifier]
[ERROR]   28. ['UnaryPrefixOperator']
[ERROR]   29. ['DecimalLiteral']
[ERROR]   30. ['HexLiteral']
[ERROR]   31. ['OctalLiteral']
[ERROR]   32. ['BinaryLiteral']
[ERROR]   33. ['FloatLiteral']
[ERROR]   34. ['HexFloatLiteral']
[ERROR]   35. ['true']
[ERROR]   36. ['false']
[ERROR]   37. ['CharLiteral']
[ERROR]   38. ['TextBlock']
[ERROR]   39. ['StringLiteral']
[ERROR]   40. ['null']
[ERROR]   41. ['this']
[ERROR]   42. ['void']
[ERROR]   43. ['byte']
[ERROR]   44. ['short']
[ERROR]   45. ['int']
[ERROR]   46. ['long']
[ERROR]   47. ['char']
[ERROR]   48. ['float']
[ERROR]   49. ['double']
[ERROR]   50. ['boolean']
[ERROR]   51. ['@']
[ERROR]   52. ['super']
[ERROR]   53. ['new']
[ERROR]   54. ['switch']
[ERROR]   55. ['assert']
[ERROR]   56. ['do']
[ERROR]   57. ['break']
[ERROR]   58. ['continue']
[ERROR]   59. ['return']
[ERROR]   60. ['synchronized']
[ERROR]   61. ['throw']
[ERROR]   62. ['try']
[ERROR]   63. ['if']
[ERROR]   64. ['while']
[ERROR]   65. ['for']
[ERROR] but found: 'static'!
[ERROR] 	->compilationUnit
[ERROR] 	->ordinaryCompilationUnit
[ERROR] 	->typeDeclaration
[ERROR] 	->classDeclaration
[ERROR] 	->normalClassDeclaration
[ERROR] 	->classBody
[ERROR] 	->classBodyDeclaration
[ERROR] 	->classMemberDeclaration
[ERROR] 	->methodDeclaration
[ERROR] 	->methodBody
[ERROR] 	->block
[ERROR] 	->blockStatements
[ERROR] 	->blockStatement
[ERROR]     at Object.parse (/home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/java-parser/src/index.js:40:11)
[ERROR]     at Object.parse (/home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/prettier-plugin-java/src/parser.js:6:26)
[ERROR]     at Object.parse$a [as parse] (/home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/prettier/index.js:14892:19)
[ERROR]     at coreFormat (/home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/prettier/index.js:16425:16)
[ERROR]     at formatWithCursor$1 (/home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/prettier/index.js:16665:14)
[ERROR]     at /home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/prettier/index.js:60922:12
[ERROR]     at Object.format (/home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/prettier/index.js:60942:12)
[ERROR]     at /home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/serve.js:58:37
[ERROR]     at Layer.handle [as handle_request] (/home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/express/lib/router/layer.js:95:5)
[ERROR]     at next (/home/jqno/w/equalsverifier/equalsverifier/target/spotless-node-modules-prettier-format/node_modules/express/lib/router/route.js:137:13))
```

--- a/packages/java-parser/src/productions/classes.js
+++ b/packages/java-parser/src/productions/classes.js
@@ -850,7 +850,8 @@ function defineRules($, t) {
 
       if (
         tokenMatcher(nextTokenType, t.Class) ||
-        tokenMatcher(nextTokenType, t.Enum)
+        tokenMatcher(nextTokenType, t.Enum) ||
+        tokenMatcher(nextTokenType, t.Record)
       ) {
         return classBodyTypes.classDeclaration;
       }
@@ -923,6 +924,7 @@ function defineRules($, t) {
   $.RULE("isCompactConstructorDeclaration", () => {
     $.MANY($.constructorModifier);
     $.SUBRULE($.simpleTypeName);
+    $.CONSUME(t.LCurly);
   });
 }
 

--- a/packages/java-parser/test/records/records-spec.js
+++ b/packages/java-parser/test/records/records-spec.js
@@ -25,4 +25,63 @@ describe("The Java Parser fixed bugs", () => {
     `;
     expect(() => javaParser.parse(input, "compilationUnit")).to.not.throw();
   });
+
+  it("should handle Java records with simplified constructors inside java class declaration", () => {
+    const input = `
+    public class RecordClass {
+      record MyRecord(String name, int age) {
+          public MyRecord {
+            if (age < 0) {
+              throw new IllegalArgumentException("Age cannot be negative");
+            }
+    
+            if (name == null || name.isBlank()) {
+              throw new IllegalArgumentException("Name cannot be blank");
+            }
+          }
+      }
+    }
+    `;
+    expect(() => javaParser.parse(input, "compilationUnit")).to.not.throw();
+  });
+
+  it("should handle Java records with constructor inside java class declaration", () => {
+    const input = `
+    public class RecordClass {
+      record MyRecord(String name, int age) {
+          public MyRecord(String name, int age) {
+            if (age < 0) {
+              throw new IllegalArgumentException("Age cannot be negative");
+            }
+    
+            if (name == null || name.isBlank()) {
+              throw new IllegalArgumentException("Name cannot be blank");
+            }
+          }
+      }
+    }
+    `;
+    expect(() => javaParser.parse(input, "compilationUnit")).to.not.throw();
+  });
+
+  it("should handle Java records with constructor inside java class declaration", () => {
+    const input = `
+    public class RecordClass {
+      public record MyRecord(String name, int age) {
+          @Annotation
+          @Annotation2
+          public MyRecord {
+            if (age < 0) {
+              throw new IllegalArgumentException("Age cannot be negative");
+            }
+    
+            if (name == null || name.isBlank()) {
+              throw new IllegalArgumentException("Name cannot be blank");
+            }
+          }
+      }
+    }
+    `;
+    expect(() => javaParser.parse(input, "compilationUnit")).to.not.throw();
+  });
 });

--- a/packages/prettier-plugin-java/scripts/clone-samples.js
+++ b/packages/prettier-plugin-java/scripts/clone-samples.js
@@ -48,7 +48,8 @@ const jhipster1 = [
   },
   {
     repoUrl: "https://github.com/jhipster/jhipster-sample-app-nocache",
-    branch: "main"
+    branch: "main",
+    commitHash: "fda6dee77651042083780c812a1a54ff2de8da47"
   },
   {
     repoUrl: "https://github.com/jhipster/jhipster-sample-app-hazelcast",
@@ -97,10 +98,21 @@ fs.emptyDirSync(samplesDir);
 
 sampleRepos.forEach(cloneRepo);
 
-function cloneRepo({ repoUrl, branch }) {
+function cloneRepo({ repoUrl, branch, commitHash }) {
   console.log(`cloning ${repoUrl}`);
-  cp.execSync(`git clone ${repoUrl} --branch ${branch} --depth 1`, {
-    cwd: samplesDir,
-    stdio: [0, 1, 2]
-  });
+  if (commitHash) {
+    cp.execSync(`git clone ${repoUrl} --branch ${branch}`, {
+      cwd: samplesDir,
+      stdio: [0, 1, 2]
+    });
+    cp.execSync(`git checkout ${commitHash}`, {
+      cwd: path.resolve(samplesDir, repoUrl.split("/").pop()),
+      stdio: [0, 1, 2]
+    });
+  } else {
+    cp.execSync(`git clone ${repoUrl} --branch ${branch} --depth 1`, {
+      cwd: samplesDir,
+      stdio: [0, 1, 2]
+    });
+  }
 }

--- a/packages/prettier-plugin-java/test/unit-test/records/_input.java
+++ b/packages/prettier-plugin-java/test/unit-test/records/_input.java
@@ -38,4 +38,55 @@ class T {
     void t() {
         record = "12";
     }
+
+class MyRecordSimplifiedConstructor {
+    record MyRecord(String name, int age
+    ) {
+public MyRecord {
+            if (age < 0) {
+              throw new IllegalArgumentException("Age cannot be negative");
+            }
+    
+      if (name == null || name.isBlank()) {
+              throw new IllegalArgumentException("Name cannot be blank");
+       }
+          }
+      }
+}
+
+class MyRecordConstructor {
+    record MyRecord(String name,
+     int age) {
+          public MyRecord(String name,
+           int age) {
+            if (age < 0) {
+  throw new IllegalArgumentException("Age cannot be negative");
+            }
+                if (name == null || name.isBlank()) {
+              throw new IllegalArgumentException("Name cannot be blank");
+        }
+          }
+      }
+}
+
+public class MyRecordWithAnnotationAndModifiers {
+                public record MyRecord(
+          String name, 
+          int age) {
+                @Annotation
+          @Annotation2
+          public MyRecord {
+            if (
+                age < 0) 
+            {
+              throw new IllegalArgumentException("Age cannot be negative");
+            }
+    
+            if (name == null || 
+            name.isBlank()) {
+              throw new IllegalArgumentException("Name cannot be blank");
+            }
+          }
+      }
+    }
 }

--- a/packages/prettier-plugin-java/test/unit-test/records/_output.java
+++ b/packages/prettier-plugin-java/test/unit-test/records/_output.java
@@ -31,4 +31,50 @@ class T {
   void t() {
     record = "12";
   }
+
+  class MyRecordSimplifiedConstructor {
+
+    record MyRecord(String name, int age) {
+      public MyRecord {
+        if (age < 0) {
+          throw new IllegalArgumentException("Age cannot be negative");
+        }
+
+        if (name == null || name.isBlank()) {
+          throw new IllegalArgumentException("Name cannot be blank");
+        }
+      }
+    }
+  }
+
+  class MyRecordConstructor {
+
+    record MyRecord(String name, int age) {
+      public MyRecord(String name, int age) {
+        if (age < 0) {
+          throw new IllegalArgumentException("Age cannot be negative");
+        }
+        if (name == null || name.isBlank()) {
+          throw new IllegalArgumentException("Name cannot be blank");
+        }
+      }
+    }
+  }
+
+  public class MyRecordWithAnnotationAndModifiers {
+
+    public record MyRecord(String name, int age) {
+      @Annotation
+      @Annotation2
+      public MyRecord {
+        if (age < 0) {
+          throw new IllegalArgumentException("Age cannot be negative");
+        }
+
+        if (name == null || name.isBlank()) {
+          throw new IllegalArgumentException("Name cannot be blank");
+        }
+      }
+    }
+  }
 }

